### PR TITLE
add `getOperation` syscall and fix tests

### PR DESCRIPTION
**Pull request by roaminro**
## Brief description
Content of this PR:

- adds `getOperation` system call (tests are written, but MockVM package needs to be updated in order to handle the new system call properly)
- makes the `defaultValue` optional in the Storage helper 
-  fixes the tests for the Storage helper (and add coverage tests)

## Checklist

- [x] I have built this pull request locally
- [x] I have ran the unit tests locally
- [x] I have manually tested this pull request
- [x] I have reviewed my pull request
- [x] I have added any relevant tests

### CODE REVIEW

File: assembly/util/mockVM.ts [hunk 1]
```diff
@@ -155,6 +155,30 @@ export namespace MockVM {
     System.putObject(METADATA_SPACE, 'transaction', transaction, protocol.transaction.encode);
   }
 
+  /**
+    * Set transaction that will be used when calling System.getTransaction() and System.getTransactionField(...)
```
Comment (sgerbino) [inline assembly/util/mockVM.ts:159]: We should probably update this comment to reflect `getOperation()` looks like it was copied from `getTransaction()` and not updated.

File: __tests__/mockVM.spec.ts [hunk 2]
```diff
@@ -78,6 +78,18 @@ describe('MockVM', () => {
     expect(Arrays.equal(getTransaction.id, setTransaction.id)).toBe(true);
   });
 
+  xit('should set the operation', () => {
```
Comment (sgerbino) [inline __tests__/mockVM.spec.ts:81]: Is `xit` a typo? Should this be `it`?

File: __tests__/systemCalls.spec.ts [hunk 3]
```diff
@@ -55,6 +55,18 @@ describe('SystemCalls', () => {
     expect(Arrays.equal(getTransaction!.bytes_value, setTransaction.id)).toBe(true);
   });
 
+  xit('should get the operation', () => {
```
Comment (sgerbino) [inline __tests__/systemCalls.spec.ts:58]: Should this also be `it`?

**Review by sgerbino — CHANGES_REQUESTED** [review]

File: assembly/util/mockVM.ts [hunk 4]
```diff
@@ -210,37 +234,37 @@ export namespace MockVM {
...
+   * ```
+   */
+  export function setVerifyVRFPRoofResults(verifyVRFProofResults: bool[]): void {
+    const verifyVRFProofResultListType = new value.list_type();
 
-        verifyVRFProofResultListType.values.push(callVerufyVRFProofValueType);
-      }
+    for (let index = 0; index < verifyVRFProofResults.length; index++) {
+      const callVerufyVRFProofValueType = new value.value_type();
+      callVerufyVRFProofValueType.bool_value = verifyVRFProofResults[index];
```
Comment (sgerbino) [inline assembly/util/mockVM.ts:260]: `Verufy` -> `Verify`

File: assembly/util/mockVM.ts (hunk 1 above)
Comment (roaminro) [reply assembly/util/mockVM.ts]: updated accordingly

File: assembly/util/mockVM.ts (hunk 4 above)
Comment (roaminro) [reply assembly/util/mockVM.ts]: updated accordingly

**Comment by roaminro** [comment]
Thanks for your review @sgerbino!
I made the needed changes and also updated the `MockVM` package and enabled the unit tests for the new `getOperation` syscall.

**Review by sgerbino — APPROVED** [review]

**Pull request merged**